### DISCUSSION TITLE
Kratos mesh I/O

### DISCUSSION THREAD

**Discussion by AdelineHunter** [KratosMultiphysics/Kratos#10175] [General]
Hello, Dear Prof.

Can Kratos support others mesh files besides `.mdpa`.

Thanks

**Comment by RiccardoRossi** [comment]
Dear Adeline,

Kratos can be started from hdf files (see the hdf5 app).

if you want other formats there is some conversion support in the package meshio

**Reply by loumalouomega** [reply]
Support from meshio is very primitive (it was me who implemented) without submodelparts and assuming certain geometries were conditions or elements. Should be improved.

We can also add support for other formats, but that takes time...

**Reply by rubenzorrilla** [reply]
See [this](https://github.com/nschloe/meshio/pull/1337). There is an ongoing PR to support submodelparts in MeshIO. This enhances usability a lot.

**Reply by AdelineHunter** [reply]
Hi RiccardoRossi, Loumaloumega, rubenzorrilla
Thank you for your reply,
In fact, if the mesh file can be supported more and more by Kartos, Kartos will be another OpenFoam in the FEM domain.
I have another question about documentation of Kartos. Is there any documentation about the reference article and how to come true all the FEM algorithms such as SUPG and Crosswind diffusion? Maybe the documentation can be called theory documentation but it contains more FEM algorithm details.
Looking forward to your reply.

**Reply by rubenzorrilla** [reply]
Hi again @AdelineHunter,

I should admittedly say that documentation has been always our weakest point... We have no theory manual. Closest thing to a theory manual would be the research papers we publish and the comments we put throughout the code. During the past years we've been a bit picky with code documentation, so latest implementations code is fairly well documented.

If you're interested in a particular application or physics the easiest way is to put you in contact with the main developers of that part of Kratos. They will point you to that specific part of the code and the papers related as well as to the corresponding tests and examples from the Examples repository.

**Reply by AdelineHunter** [reply]
Hi, rubenzorrilla

That is a good idea, I am interested in the CFD module with the FEM method, especially the two-equations RANS turbulence model...
So, do you have any discussion on the robustness of Kartos' RANS  application? Or are there any more test cases about the RANS application?

Thank you for your reply again.

**Reply by rubenzorrilla** [reply]
For any fluid-related query you can ping all the fluid maintainers (@KratosMultiphysics/fluid-maintainers) or directly me, @RiccardoRossi and @sunethwarna. However, if you've particular interest on the RANS capabilities, @sunethwarna is the man.

(@sunethwarna see above).

**Reply by sunethwarna** [reply]
Hi @AdelineHunter,

The `RANSApplication` is under development and it can be used at its current state. I've been doing more enhancements to it in my branch (yet to be merged to the master). You can check it out in https://github.com/KratosMultiphysics/Kratos/tree/rans/feature/adjoint/vortex_shedding_frequency_sensitivities.

Unfortunately as @rubenzorrilla points out, there are no documentation for that yet. We are working on it. If you have question, I am happy to answer them.

**Reply by AdelineHunter** [reply]
Hi, Prof. @sunethwarna 

Thanks for your reply.

Your work on developing a RANS application is exciting.  I am also interested in the CFD module with the FEM method, especially the two-equations RANS turbulence model... But unfortunately, I am confused and depressed by the convergence problem in my FEM program.

As far as I know, SUPG, VMS, Crosswind Diffusion stabilization and etc are introduced in the Kratos program. I did know that SUPG is essential to the simulation. But I do not know VMS, Crosswind Diffusion stabilization and etc is or not essential.

Could you please introduce some reference articles or books which you had read when you program the RANS application and which you think highly of?

Best wishes.

**Reply by sunethwarna** [reply]
Hi @AdelineHunter 

btw, I'm not a prof. ;). 

As for your questions, 

> As far as I know, SUPG, VMS, Crosswind Diffusion stabilization and etc are introduced in the Kratos program. I did know that SUPG is essential to the simulation. But I do not know VMS, Crosswind Diffusion stabilization and etc is or not essential.

VMS also has SUPG stabilization within it, it can be seen as a different way of modelling small scales which are not resolved. In the case of RANS, the stabilization should be positivity preserving. Therefore we have implemented `AlgebraicFluxCorrection` method (https://link.springer.com/chapter/10.1007/3-540-27206-2_6) `CrosswindDiffusion` method (DOI: 10.1016/j.jcp.2017.03.005) and `ResidualBasedFluxCorrection` (paper under review yet). May be you can give them a try. 

Best regards,
Suneth
